### PR TITLE
Update tutorial builtin contract list

### PR DESCRIPTION
**Pull request by gsinclair**
There are three commits in this pull request. One to add Eq and Func. One to arrange the list into groups. One to make a few changes to wording and examples.

This addresses the documentation deficiency identified in #168.

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -67,26 +67,39 @@ This can be useful if you're in a REPL and want to figure out how a function sho
...
+* Collections
+  * [`ArrayOf`](http://www.rubydoc.info/gems/contracts/Contracts/ArrayOf) – checks that the argument is an array, and all elements pass the given contract, e.g. `ArrayOf[Num]`
+  * [`SetOf`](http://www.rubydoc.info/gems/contracts/Contracts/SetOf) – checks that the argument is a set, and all elements pass the given contract, e.g. `SetOf[Num]`
+  * [`HashOf`](http://www.rubydoc.info/gems/contracts/Contracts/HashOf) – checks that the argument is a hash, and all keys and values pass the given contract, e.g. `HashOf[Symbol => String]` or `HashOf[Symbol,String]`
+
+* Keyword arguments
+  * [`KeywordArgs`](http://www.rubydoc.info/gems/contracts/Contracts/KeywordArgs) – checks that the argument is an options hash, and all required keyword arguments are present, and all values pass their respective contracts, e.g. `KeywordArgs[:number => Num, :description => Optional[String]]`
+  * [`Optional`](http://www.rubydoc.info/gems/contracts/Contracts/Optional) – checks that the keyword argument is either not present or pass the given contract, can not be used outside of `KeywordArgs` contract, e.g. `Optional[Num]`
+
+* Validators
```
Comment (alex-fedorov) [inline TUTORIAL.md]: Maybe `Duck typing` ? WDYT?

**Comment by alex-fedorov** [comment]
Looks good

**Comment by gsinclair** [comment]
"Duck typing" sounds quite reasonable.

**Comment by waterlink** [comment]
This PR needs to be rebased against master (last merge to master has fixed issue on travis)

**Comment by gsinclair** [comment]
OK. Can you give me advice on how to do this? I'm fine with rebasing _within_ a branch, but I'm not sure how to rebase an existing branch onto master. I've googled and there's information, so that's no problem, but wouldn't I then need to `push --force` the branch? And isn't that a sin in a public repository?

**Comment by nixpulvis** [comment]
Force pushing a public branch is generally not a good idea, but only if other people are using it. So in this case you could rebase this branch against master and force push it because it's not a publicly used branch.

**Comment by gsinclair** [comment]
Cool, thanks.

**Comment by egonSchiele** [comment]
Thanks for these PRs @gsinclair! Just waiting for the rebase before I merge this.

**Comment by gsinclair** [comment]
OK, I hope I've done it right. Can you check?

**Comment by waterlink** [comment]
@gsinclair Doesn't look like you have successfully rebased against upstream master, or probably something else, that conflicts with this change was merged in between.

Just in case, my usual rebase procedure: https://gist.github.com/waterlink/ee7600e499d2ca8c596f

**Comment by gsinclair** [comment]
Thanks for your procedure; it was helpful.

I've followed it, resolved conflicts, and force pushed my topic branch again. Should be OK now, I'd say.

Note: my branch has five commits; please feel free to squash them into one before merging. I'd happily do that for you, but it's not my call.

**Comment by waterlink** [comment]
@gsinclair Thanks

**Pull request merged**